### ISSUE TITLE
Test: GitHub Pull Request support

### ISSUE THREAD

**Issue by rebornix** [Microsoft/vscode-pull-request-github#15]
- [x] Windows @alexandrudima 
- [x] Linux @Tyriar 
- [x] macOS @joaomoreno 

Complexity: 5

This extension is in preview state and we didn't cover all features and didn't have a sound solution for multi-root workspace or authentication yet. We can talk about features/ideas/must-haves in our roadmap https://github.com/Microsoft/vscode-pull-request-github/issues/1 and log issues for any bug, stability problem, blockers you run into while testing the extension.

Obvious gaps that we'll close over time:
- No multi-root folder support
- No assignees / reviewers
- No support for the GH Review flow
- No explicit merge support

## Setup
* Download the latest visx from our release page
* Install the vsix in latest Insider
* Run Insider from command with flag `code-insiders --enable-proposed-api Microsoft.vscode-pull-request-github`

The last step is necessary as we didn't add this extension to the proposed api whitelist. You'll see corresponding errors if you didn't pass in the right flag.

**Authentication**

There are two ways of doing authentication

* GitHub access token, this is the recommended way right now. The config name is `github.accessToken`. Please make sure the token you put there has proper access to *repo* and *user*
* If there is no GitHub access token, the extension will try to use the local Git Credential Manager. Whether the authentication will work depends on if your local credentials work.
  * There is an known bug when you are using ssh and the `passphrase` expires, the extension will fail slightly. 

## Testing

You may want to test at least following scenarios:

**Pull Request tree view**:
* View pull requests in SCM viewlet when a GitHub repository is opened in VSCode, the predefined categories work as expected.
* View file changes from the Pull Request tree view
* View/create comments when viewing file changes from the Pull Request tree view
* View the description page for a pull request, the timeline events of a pull request are rendered properly
* Leave generic issue comments in the description page for a pull request
* Context menu for a pull request item, file change item in the tree view

**Review mode**
* Checkout to a pull request from Pull Request description page or content menu of Pull Request tree view
* Comments for the pull request are loaded and a comment panel shows up accordingly.
* Navigate through comments in the Command Panel
* View/Create comments in a normal editor
* Comments positions are adjusted when you make changes to local content.

**Comment by alexandrudima** [comment]
Sorry, I could not get it to work on Windows.

After I manually ran `yarn` in the extension directory (to work around #17), I am running all the time into #32 and it doesn't look like I can get out of that one.

**Comment by rebornix** [comment]
@alexandrudima thanks, I didn't realize that dugite ships git and requires bundling for every platform. Published a new version without dugite https://github.com/Microsoft/vscode-pull-request-github/releases/tag/v0.0.2 . Tested on all platforms and didn't see any issue when directly install the new vsix, but if you run into any weird issue (like the tree view is registered but a command is missing), removing the extension folder and then reinstall can work.

I also uploaded windows specific VSIX with the dugite embedded for the first release, but it's probably not needed any more.
